Script to parse recursively cms-sw/cmsdist spec files

mapellidario opened the pull request:
Fixes #10035

#### Status

ready

#### Description

We provide a script that parses a spec file and produces a requirements.txt files with all the python packegs in the spec file itself and all its dependencies

#### Is it backward compatible (if not, which system it affects?)

yes

#### External dependencies / deployment changes

This can be run from our laptop, so it does not require any change in the build and deploy process

mapellidario committed:
Script to parse recursively cms-sw/cmsdist spec files

amaltaro reviewed (commented):
Dario, from our discussion in one of the WMCore meetings. I think it was suggested not to get dependencies recursively. Should we update this script to only consider top level python libraries and then provide your procedure to deploy those services and get a `pip freeze` with all the inner dependencies?

mapellidario commented:
I updated the script so that by default it is not recursive. 

I would still keep the recursive behavior, it reminded me that the current `wmagentpy3.spec` still indirectly depends on `nats-client` (from `cmsmonitoring`), `restkit` (from `couchapp`) and `python-cjson` (from `dbs3-pycurl-client.spec`).

Moreover, I added some functions to guess the package name from `%setup`, `%define pip_name` or `%define downloadn` spec files directives. Manual changes to the output requirements.txt should not be necessary anymore.

amaltaro commented:
> I would still keep the recursive behavior, it reminded me that the current wmagentpy3.spec still indirectly depends on nats-client (from cmsmonitoring), restkit (from couchapp) and python-cjson (from dbs3-pycurl-client.spec).

We'll cross that bridge once we get to it ;)

Thanks Dario.

amaltaro commented:
Actually, please take care of all the pylint/pep issues reported. New files should come as clean as possible.

amaltaro commented:
Transient problem with DBS. Thanks for sorting all of those Dario.
